src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -175,11 +175,14 @@ def __new__(cls, val):
         return str.__new__(cls, val)
 
 
-def resource_exists(cli_ctx, resource_group, name, namespace, type, **_):  # pylint: disable=redefined-builtin
+def resource_exists(cli_ctx, subscription, resource_group, name, namespace, type,
+                    **_):  # pylint: disable=redefined-builtin
```

Comment:
Have you confirmed that inserting `subscription` in the middle of the parameter list will not cause parameter misalignment issues in previous calls?

Reply:
Please note that this is a method in the azure-cli-core, so we also need to consider the invocation of other external dependencies such as CLI extensions. If you are not 100% confident, it is recommended to insert `subscription` parameter at the end of the parameter list.

Reply:
Yes, only two functions(`get_folded_parameter_validator(core/commands/templated_create.py)` and `_resource_not_exists(vm/_actions.py)`) in azure-cli and azure-cli-extensions use this function and I have update the call in `_resource_not_exists` correspondingly.